python/lsst/ts/scheduler/model.py

```diff
@@ -1,44 +1,132 @@
+            middleware communication capabilities.
+        configuration_path (str): Directory path for configuration files.
+        configuration_repo (str): Comma delimmited string of available branches
+            for configuring. This contains the full branch name.
+        current_setting (str): Current setting, defaults to the active branch on
+            the configuration_repo.
+        valid_settings (str): Comma delimmited string of available branches
+            for configuring. This contains the short branch name. For example
+            'master' rather than 'origin/master'.
+
     """
+    def __init__(self, driver = Driver()):
+        """The Model object is planned to be configurable by a method TODO:
+
+        Note:
+            The Model is the highest level interface for the State Machine to interact
+            with. The Scheduler State Machine has multiple states. Rather than having
+            to import many objects from different locations we import this Model object.
+            Allowing the Scheduler State Machine one unified object to make all its
+            calls from.
+
+        Args:
+            driver (:obj:`Driver()`, optional): Externally provided Drivers must
+                comply with the Driver interface here
+                https://docs.google.com/document/d/1wQ467QGqVP2e4iXOKcW2slLMgOzI_1Qk284b1gDGqa.
+                Current Driver can be found in lsst.ts.scheduler.Driver
+
+        """
+        self._current_state = DEFAULT_STATE
+        self._previous_state = DEFAULT_STATE
+
+        self.log = logging.getLogger("schedulerModel")
+        self.sal = SALUtils(2000)
+        self.configuration_path = str(CONFIG_DIRECTORY)
+        self.configuration_repo = Repo(str(CONFIG_DIRECTORY_PATH))
+        self.current_setting = str(self.configuration_repo.active_branch)
+        self.valid_settings = self.read_valid_settings()
+
+    def sal_start(self):
+        """Initializes SAL communication.
 
-    def __init__(self):
+        Note:
+            SAL communication is not initialized within our init function so
+            that we may test various methods that do not require SAL
+            communication.
 
-        self.state = DEFAULT_STATE
-        self.previous_state = self.state
+        """
+        self.sal.start()
 
-        self.mgr = SAL_scheduler()
+    def read_valid_settings(self):
+        """Reads the branches on the configuration repo and preps them.
 
-    def change_state(self, state):
+        Returns:
+            List of branches on the configuration repository. A single branch
+            represents a valid setting.
 
-        self.previous_state = self.state
-        self.state = state
+        """
+        remote_branches = []
+        for ref in self.configuration_repo.git.branch('-r').split('\n'):
+            if 'HEAD' not in ref:
+                remote_branches.append(ref)
+
+        return remote_branches
+
+    @property
+    def current_state(self):
```

Comment:
This is pretty neat! It really nice to be able to simply do `model.current_state = 'STANDBY'` and it handles it all.